File: code/58.length-of-last-word.java
/*
 * @lc app=leetcode id=58 lang=java
 *
 * [58] Length of Last Word
 */

// @lc code=start
class Solution {
    public int lengthOfLastWord(String s) {
        int count = 0;
        for (int i = s.length() - 1; i >= 0; i--) {
            if (isAlphabet(s.charAt(i))) {
                count++;
            } else if (count > 0) {
                break;
            }
        }
        return count;
    }

    public boolean isAlphabet(char a) {
        if ((a <= 'z' && a >= 'a') || (a <= 'Z' && a >= 'A')) {
            return true;
        }
        return false;
    }
}
// @lc code=end

File: code/58.length-of-last-word.go
/*
 * @lc app=leetcode id=58 lang=golang
 *
 * [58] Length of Last Word
 */

// @lc code=start
func lengthOfLastWord(s string) int {
	count := 0
	for i := len(s) - 1; i >= 0; i-- {
		if isAlphabet(s[i]) {
			count++
		} else if count > 0 {
			break
		}
	}
	return count
}

func isAlphabet(c byte) bool {
	return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z')
}

// @lc code=end

